### DISCUSSION TITLE
Is Axum FIPS compliant ?

### DISCUSSION THREAD

**Discussion by sarmahaj** [tokio-rs/axum#2134] [Q&A]
We currently use warp web-frame work but it uses `rustls` and it uses `ring` crate which is not FIPs compliant. So looking for moving to Axum . But before that would like to confirm if Axum complies FIPS ( Federal Information Processing Standards )
Thanks !

**Comment by davidpdrsn** [comment]
No idea what that is so likely not 😅

**Comment by sarmahaj** [comment]
FIPS is basically US gov standards for cryptographic related crates. 
Thanks for the reply. Just on that note if I could get more info on using axum to support https connections that would be great! If it supports native-tls for TLS config ?

**Reply by davidpdrsn** [reply]
Well axum isn't a cryptographic crate and doesn't have TLS support built-in, you have to handle that yourself. See the examples in the repo but they all use `rustls` afaik.
